rest/data/commit_queue.go

```diff
@@ -69,19 +73,90 @@ func (pc *DBCommitQueueConnector) FindCommitQueueByID(id string) (*restModel.API
 	return apiCommitQueue, nil
 }
 
-func (pc *DBCommitQueueConnector) CommitQueueRemoveItem(id, item string) (bool, error) {
+func (pc *DBCommitQueueConnector) CommitQueueRemoveItem(id, item string) (*commitqueue.CommitQueueItem, error) {
 	cq, err := commitqueue.FindOneId(id)
 	if err != nil {
-		return false, errors.Wrapf(err, "can't get commit queue for id '%s'", id)
+		return nil, errors.Wrapf(err, "can't get commit queue for id '%s'", id)
 	}
 
-	return cq.Remove(item)
+	head := cq.Next()
```

Comment:
I believe it's because we only need to prevent merge if the item we want to remove is at the head of the queue (i.e. being merged), whereas if it's later in the queue there is nothing to prevent as long as it's removed.

Reply:
Maybe it would be clearer if this was head.Issue == itemRemoved.Issue?

Reply:
no longer possible (reverted to returning a bool)